src/commonMain/kotlin/kotools/types/collection/NotEmptySet.kt

```diff
@@ -55,6 +58,49 @@ public fun <E> Collection<E>.toNotEmptySet(): Result<NotEmptySet<E>> =
         }
         ?: Result.failure(EmptyCollectionException)
 
+/**
+ * Representation of sets that contain at least one element of type [E].
+ *
+ * You can use the [notEmptySetOf] or the [toNotEmptySet] functions for building
+ * this type.
+ */
+@JvmInline
+@Serializable(NotEmptySetSerializer::class)
+@SinceKotoolsTypes("4.0")
+public value class NotEmptySet<out E> private constructor(
+    private val elements: Set<E>
+) : NotEmptyCollection<E> {
+    override val head: E get() = elements.first()
+    override val tail: NotEmptySet<E>?
+        get() = elements.drop(1)
+            .toNotEmptySet()
+            .getOrNull()
+
+    internal constructor(head: E, tail: NotEmptySet<E>? = null) : this(
+        tail?.run { setOf(head) + elements }
+            ?: setOf(head)
+    )
+
+    /**
+     * Returns all elements of this set as a [Set] of type [E].
+     *
+     * ```kotlin
+     * import kotools.types.collection.NotEmptySet
+     * import kotools.types.collection.notEmptySetOf
+     *
+     * val notEmptySet: NotEmptySet<Int> = notEmptySetOf(1, 2, 3, 1)
+     * val set: Set<Int> = notEmptySet.toSet()
+     * println(set) // [1, 2, 3]
+     * ```
+     */
+    public fun toSet(): Set<E> {
+        val elements: Set<E> = setOf(head)
+        return tail?.let { elements + it.toSet() } ?: elements
+    }
```

Comment:
We can return directly the `elements` property instead.

```suggestion
    public fun toSet(): Set<E> = elements
```

Reply:
Resolved in the commit b1776ef76925a42fabda0de542ae1b675a5df1bc.